app/services/laa_reference_updater.rb

```diff
@@ -6,17 +6,19 @@ def initialize(contract)
   end
 
   def call
-    maat_reference = contract[:maat_reference]
+    @maat_reference = contract[:maat_reference] || dummy_maat_reference
 
     ProsecutionCaseDefendantOffence.where(defendant_id: contract[:defendant_id]).each do |offence|
-      Api::RecordLaaReference.call(
+      @response = Api::RecordLaaReference.call(
         prosecution_case_id: offence.prosecution_case_id,
         defendant_id: offence.defendant_id,
         offence_id: offence.offence_id,
         status_code: 'AP',
-        application_reference: maat_reference || dummy_maat_reference,
+        application_reference: @maat_reference,
         status_date: Date.today.strftime('%Y-%m-%d')
       )
+
+      update_record(offence)
```

Comment:
Could this be better suited to be triggered from within the `Api::RecordLaaReference` call?
similar to [the get hearing call](https://github.com/ministryofjustice/laa-court-data-adaptor/blob/1777cd4b39f9baeac8d28d4e157cb98c21891956/app/services/api/get_hearing_results.rb)

Reply:
Moved to `Api::RecordLaaReference`